cli/internal/packager/seed.go

```diff
@@ -0,0 +1,247 @@
+
+	embeddedRegistry, err := registry.NewRegistry(ctx, registryConfig)
+	if err != nil {
+		message.Fatal(err, "Unable to start the embedded registry")
+	}
+
+	//go func() {
+	if err := embeddedRegistry.ListenAndServe(); err != nil {
+		message.Fatal(err, "Unable to start the embedded registry")
+	}
+	//}()
+
+	stopSeedRegistry = done
+}
+
+func preSeedRegistry(tempPath tempPaths) {
+	message.Debugf("package.preSeedRegistry(%v)", tempPath)
+
+	var (
+		distro string
+		err    error
+		inject struct {
+			command string
+			args    []string
+		}
+	)
+
+	// Attempt to load an existing state prior to init
+	state := k8s.LoadZarfState()
+
+	if state.Secret == "" || state.Distro == k8s.DistroIsUnknown {
+		// If the state is invalid, assume this is a new cluster
+		message.Debug("New cluster, no zarf state found")
+
+		if config.DeployOptions.ApplianceMode {
+			// If the K3s component is being deployed, skip distro detection
+			distro = k8s.DistroIsK3s
+			state.ZarfAppliance = true
+		} else {
+			// Otherwise, trying to detect the K8s distro type
+			distro, err = k8s.DetectDistro()
+			if err != nil {
+				// This is a basic failure right now but likely could be polished to provide user guidance to resolve
+				message.Fatal(err, "Unable to connect to the k8s cluster to verify the distro")
+			}
+		}
+
+		message.Debugf("Detected K8s distro %v", distro)
+
+		// Defaults
+		state.Registry.NodePort = "31999"
+		state.Secret = utils.RandomString(120)
+		state.Distro = distro
+		state.Architecture = config.GetBuildData().Architecture
+	}
+
+	switch state.Distro {
+	case k8s.DistroIsK3s:
+		state.StorageClass = "local-path"
+		state.Registry.SeedType = config.ZarfSeedTypeCLIInject
+		inject.command = "k3s"
+		inject.args = []string{"ctr", "images", "import", tempPath.seedImages}
+
+	case k8s.DistroIsK3d:
+		state.StorageClass = "local-path"
+		clusterName := getClusterName("k3d")
+		state.Registry.SeedType = config.ZarfSeedTypeCLIInject
+		inject.command = "k3d"
+		inject.args = []string{"images", "import", tempPath.seedImages, "--cluster", clusterName}
+
+	case k8s.DistroIsKind:
+		state.StorageClass = "standard"
+		// See https://github.com/kubernetes-sigs/kind/blob/v0.11.1/pkg/cluster/internal/kubeconfig/internal/kubeconfig/helpers.go#L24
+		clusterName := getClusterName("kind")
+		state.Registry.SeedType = config.ZarfSeedTypeCLIInject
+		inject.command = "kind"
+		inject.args = []string{"load", "image-archive", tempPath.seedImages, "--name", clusterName}
+
+	case k8s.DistroIsDockerDesktop:
+		state.StorageClass = "hostpath"
+		state.Registry.SeedType = config.ZarfSeedTypeCLIInject
+		inject.command = "docker"
+		inject.args = []string{"load", "-i", tempPath.seedImages}
+
+	case k8s.DistroIsMicroK8s:
+		state.Registry.SeedType = config.ZarfSeedTypeCLIInject
+		inject.command = "microk8s"
+		inject.args = []string{"ctr", "images", "import", tempPath.seedImages}
+
+	default:
+		state.Registry.SeedType = config.ZarfSeedTypeRuntimeRegistry
```

Comment:
I could use a big more context about when/what this default state means. When we set this as the `state.Registry.SeedType` I think we are going to be running into one of our own Fatals downstream and I want to try to understand the intended logic a bit more.

@jeff-mccoy

Reply:
I'm not too worried about it (for this PR), but the currently flow will lead to a Fatal in `cli/internal/message/tls/prompts.PromptAndValidateHost()` since the `config.TLS.Host` will be an empty string but the `confirmed` flag will ALWAYS true because the user already went through the interactive confirm step.